libnetwork/osl/namespace_linux.go

```diff
@@ -692,5 +700,8 @@ func (n *networkNamespace) ApplyOSTweaks(types []SandboxType) {
 		case SandboxTypeLoadBalancer:
 			kernel.ApplyOSTweaks(loadBalancerConfig)
 		}
+		case SandboxTypeIngress:
+			kernel.ApplyOSTweaks(ingressConfig)
```

Comment:
It looks like both Ingress and Loadbalancer are applying the same tweaks. The `loadBalancerConfig` was already there, but looking at that variable, I see it's _only_ used in this function, but defined as a package level variable. I'm not a huge fan of package level variables (if not needed), especially if they're defined way too far away from where they're used.

To prevent the risk that they diverge (assuming they should always get the same tweaks - at least currently), perhaps either inline the settings we apply. The `CheckFn: nil` also seems a bit redundant (I think that'd be the default), so I assume we can remove those as well.
Finally, it looks like the description for `ApplyOSTweaks` no longer matches what it does (it should now mention the ingress network as well?);

```go
// ApplyOSTweaks applies linux configs on sandboxes and ingress networks.
func (n *networkNamespace) ApplyOSTweaks(types []SandboxType) {
	for _, t := range types {
		switch t {
		case SandboxTypeLoadBalancer, SandboxTypeIngress:
			kernel.ApplyOSTweaks(map[string]*kernel.OSValue{
				// disables any special handling on port reuse of existing IPVS connection table entries
				// more info: https://github.com/torvalds/linux/blob/master/Documentation/networking/ipvs-sysctl.txt#L25:1
				"net.ipv4.vs.conn_reuse_mode": {Value: "0"},
				// expires connection from the IPVS connection table when the backend is not available
				// more info: https://github.com/torvalds/linux/blob/master/Documentation/networking/ipvs-sysctl.txt#L126:1
				"net.ipv4.vs.expire_nodest_conn": {Value: "1"},
				// expires persistent connections to destination servers with weights set to 0
				// more info: https://github.com/torvalds/linux/blob/master/Documentation/networking/ipvs-sysctl.txt#L144:1
				"net.ipv4.vs.expire_quiescent_template": {Value: "1"},
			})
		}
	}
}
```


Or if there's advantages to defining the variable once (not sure if there is), it'd be better to move the variable closer to this function (but I'm slightly in favour of the approach above instead);

```go
var tweaks = map[string]*kernel.OSValue{
	// disables any special handling on port reuse of existing IPVS connection table entries
	// more info: https://github.com/torvalds/linux/blob/master/Documentation/networking/ipvs-sysctl.txt#L25:1
	"net.ipv4.vs.conn_reuse_mode": {Value: "0"},
	// expires connection from the IPVS connection table when the backend is not available
	// more info: https://github.com/torvalds/linux/blob/master/Documentation/networking/ipvs-sysctl.txt#L126:1
	"net.ipv4.vs.expire_nodest_conn": {Value: "1"},
	// expires persistent connections to destination servers with weights set to 0
	// more info: https://github.com/torvalds/linux/blob/master/Documentation/networking/ipvs-sysctl.txt#L144:1
	"net.ipv4.vs.expire_quiescent_template": {Value: "1"},
}

// ApplyOSTweaks applies linux configs on sandboxes and ingress networks.
func (n *networkNamespace) ApplyOSTweaks(types []SandboxType) {
	for _, t := range types {
		switch t {
		case SandboxTypeLoadBalancer, SandboxTypeIngress:
			kernel.ApplyOSTweaks(tweaks)
		}
	}
}
```

Reply:
Do you know if these are settings that can be controlled through a `com.docker.network` option on `docker network create` (https://docs.docker.com/engine/reference/commandline/network_create/#bridge-driver-options)?

Double checking if this is something that a user could have customised when creating a custom ingress network (and which we would then override with fixed defaults)

Reply:
They can't be set through `com.docker.network`, unfortunately, but I'm in total agreement about package-level variables which only get moved once. I'll inline them.